### ISSUE TITLE
Multiple parts Slice Incorrect

### ISSUE THREAD

**Issue by yuusuf-sallahuddin** [Ultimaker/Cura#11383]
### Application Version

4.13.1

### Platform

Windows 10

### Printer

Ender 3 V2

### Reproduction steps

I created a model (Rhino 3D): a drawer box, a drawer for that container, and some locking pins (to be able to lock multiple containers together).

I sliced 5mm slice of all 3 models, saved as one OBJ file. So there was a 5mm slice of the box, a 5mm slice of the drawer, a 5mm slice of the locking pins.

Upon slicing, 50mm/s, 10% infill zigzag, no supports, other default settings, it created the gcode file for printer.

I placed in printer and began printing. It began by printing on the right side of the bed, drawing a skirt around the locking pin (approx 4mm x 4mm pin, maybe 10x10mm skirt). It then printed the full 5mm slice of the locking pin.

Then it moved back to the centre of the bed and began attempting to print the skirt for the drawer slice. It was not adhering to the build plate, so I cancelled print and began recalibrating my bed level. After many hours & much frustration, I believed I was level again (equal measurements each corner & centre of bed). I began another print with the same gcode, knowing it should work now. Again, it got the the drawer slice & would not adhere.

I cancelled, cleaned up & this time turned on rafts for bed adhesion.

I watched closely on the next attempt for the print. It printed the raft & full 5mm slice of the locking pin, then went on to print the raft for the drawer slice. It was not adhering again, only this time, as there was supposed to be a raft, I noticed that the head had not lowered z back to 0.

So, after printing first "part" of the 3 parts, it did not lower the head back to 0 to begin the raft/skirt for 2nd part. Never got as far as 3rd part, but assume it would have done the same.

### Actual results

See above:

TLDR; Basically Head not returning to Z0 after printing one of 3 parts & beginning 2nd part.

### Expected results

Head should return to Z0 to begin printing the 2nd/3rd/subsequent parts.

Or, each Layer 1 of each part should be printed before moving to Layer 2 of each part
[Buggy Slicing CURA 4.13.1.zip](https://github.com/Ultimaker/Cura/files/7966043/Buggy.Slicing.CURA.4.13.1.zip)
.

### Checklist of files to include

- [ ] Log file
- [ ] Project file

### Additional information & file uploads

Nil required. All explained above. I've checked the GCODE, there is no reset of Z position after completion of 1st part printed.

**Comment by GregValiant** [comment]
The gcode for the skirt file reflects exactly how the models were sliced.  The purge lines are at .3 layer height and the rest of the print is at .2 layer height with .2 increments from layer to layer.  Unfortunately, the second and third models were not on the build plate when you sliced them.  The bottom layer of the second model and the pin are at Z=4.2.
Here is a screen shot from Cura.  I get the same thing when reading the gcode into AutoCad.
![image](https://user-images.githubusercontent.com/64202104/151704784-a966c3e8-af46-48ee-b483-e813720432b1.png)
